rules-reviewed/openshift/socket-communication.windup.xml

```diff
@@ -41,10 +43,12 @@
             </when>
             <perform>
               <iteration>
-                <hint title="Java APIs (java.nio.channels) for socket communication" effort="3" category-id="cloud-optional">
-                  <message>
-                      Avoid direct use of client sockets. Consider using Java EE standard communication features, especially loosely coupled protocols (JAX-WS, JAX-RS) instead.
-                  </message>
+                <hint title="Channel communication" effort="5" category-id="cloud-optional">
+                    <message>
+                      Direct communication through sockets and channels is an anti-pattern in cloud environment because it isn't a reliable and scalable way of interact with other systems.  
```

Comment:
Good job!

isn't -> is not

Could you add one sentence at the beginning explaining what Java NIO channels are:

"Java NIO Channels are designed to provide for bulk data transfers to and from NIO buffers. They can be synchronously and asynchronously read and written."